### ISSUE TITLE
.AppImage failing to run correctly

### ISSUE THREAD

**Issue by danrobi11** [beakerbrowser/beaker#1568]
Hello
first time launching the beaker Version: 1.0.0-prerelease.1 appimage
Beaker did found my 'site' directory (from previous beaker version)
Asking to convert to hyperdrive
I have this error instead of seeing my file: Error: Uncaught NotFoundError: File not found
Same error with all my site file.
I see alot of same error in terminal. not sure I should past the entire log here. let me know if u want me to look for something specific
```
error:    Uncaught exception: TypeError: cb is not a function {"category":"browser"}
2020-05-14T15:41:22: PM2 error: Uncaught exception: TypeError: cb is not a function
    at /tmp/.mount_BeakerAbBOgy/resources/app/node_modules/call-me-maybe/index.js:11:28
    at processTicksAndRejections (internal/process/task_queues.js:76:11)
``` 
MX Linux MX x86_64 
Kernel: 4.19.0-6-amd64 
DE: Xfce 

![error-file-not-found-beaker-beta1](https://user-images.githubusercontent.com/48020114/81980368-7a0a1600-95fc-11ea-9f56-7bb95da765d2.png)

**Comment by aadilayub** [comment]
I'm facing the same error/problem when trying to edit my profile site. I also can't edit any of the files in my profile's hyperdrive.

![image](https://user-images.githubusercontent.com/31581044/81982940-009a1400-964c-11ea-8223-e8153e466419.png)

For reference, I'm running on Ubuntu 20.04

**Comment by pfrazee** [comment]
Does this "File Not Found" error show for anywhere other than the files explorer? Does the editor work?

**Comment by pfrazee** [comment]
Also could you check `~/.hyperdrive/log.json` to see if there are any useful errors in there?

**Comment by danrobi11** [comment]
```
{"level":30,"time":1589482843698,"pid":69,"hostname":"mx","name":"hyperdrive","component":"server","msg":"memory only? false no announce? false","v":1}
{"level":30,"time":1589482843722,"pid":69,"hostname":"mx","name":"hyperdrive","component":"server","swarmId":"994e7bde5ca7e70b54955c0b0607e5a0dde22133382850315784ac0c0ca53bf1","seed":"b21a5c2b4c819c2a52047d4513893e5843475a73109eb5f1d12ecf95605bd90e","msg":"creating replication keypair and swarm ID","v":1}
{"level":50,"time":1589482843814,"pid":69,"hostname":"mx","name":"hyperdrive","component":"server","error":{"code":"fuse failed"},"stack":"Error: fuse failed\n    at /run/firejail/appimage/.appimage-1794/resources/app/node_modules/fuse-native/index.js:311:21\n    at FSReqCallback.oncomplete (fs.js:159:5)","msg":"error in daemon ready function -- cleaning up","v":1}
{"level":50,"time":1589482843814,"pid":69,"hostname":"mx","name":"hyperdrive","component":"server","error":true,"err":{"type":"Error","message":"fuse failed","stack":"Error: fuse failed\n    at /run/firejail/appimage/.appimage-1794/resources/app/node_modules/fuse-native/index.js:311:21\n    at FSReqCallback.oncomplete (fs.js:159:5)","code":"fuse failed"},"message":"fuse failed","stack":"Error: fuse failed\n    at /run/firejail/appimage/.appimage-1794/resources/app/node_modules/fuse-native/index.js:311:21\n    at FSReqCallback.oncomplete (fs.js:159:5)","msg":"stopping daemon due to error","v":1}
{"level":30,"time":1589482843814,"pid":69,"hostname":"mx","name":"hyperdrive","component":"server","msg":"waiting for fuse to unmount","v":1}
{"level":30,"time":1589482843815,"pid":69,"hostname":"mx","name":"hyperdrive","component":"server","msg":"waiting for networking to close","v":1}
{"level":30,"time":1589482843815,"pid":69,"hostname":"mx","name":"hyperdrive","component":"server","msg":"waiting for corestore to close","v":1}
{"level":30,"time":1589482843816,"pid":69,"hostname":"mx","name":"hyperdrive","component":"server","msg":"waiting for db to close","v":1}
{"level":30,"time":1589483711060,"pid":3746,"hostname":"mx","name":"hyperdrive","component":"server","msg":"memory only? false no announce? false","v":1}
{"level":30,"time":1589483711100,"pid":3746,"hostname":"mx","name":"hyperdrive","component":"server","swarmId":"994e7bde5ca7e70b54955c0b0607e5a0dde22133382850315784ac0c0ca53bf1","seed":"b21a5c2b4c819c2a52047d4513893e5843475a73109eb5f1d12ecf95605bd90e","msg":"creating replication keypair and swarm ID","v":1}
{"level":30,"time":1589483711178,"pid":3746,"hostname":"mx","name":"hyperdrive","component":"server","port":3101,"msg":"server listening","v":1}
{"level":30,"time":1589484173373,"pid":3746,"hostname":"mx","name":"hyperdrive","component":"drive-manager","key":"6900790c2dba488ca132a0ca6d7259180e993b285ede6b29b464b62453cd5c39","msg":"registering mountpoint in drive index","v":1}
{"level":50,"time":1589484397758,"pid":3746,"hostname":"mx","name":"hyperdrive","component":"drive-manager","id":4,"path":"/","err":{"wrappers_":null,"arrayIndexOffset_":-1,"array":[],"pivot_":1.7976931348623157e+308,"convertedPrimitiveFields_":{}},"msg":"watch stream errored","v":1}
{"level":50,"time":1589484399842,"pid":3746,"hostname":"mx","name":"hyperdrive","component":"drive-manager","id":4,"path":"/","err":{"wrappers_":null,"arrayIndexOffset_":-1,"array":[],"pivot_":1.7976931348623157e+308,"convertedPrimitiveFields_":{}},"msg":"watch stream errored","v":1}
{"level":50,"time":1589484403592,"pid":3746,"hostname":"mx","name":"hyperdrive","component":"drive-manager","id":4,"path":"/","err":{"wrappers_":null,"arrayIndexOffset_":-1,"array":[],"pivot_":1.7976931348623157e+308,"convertedPrimitiveFields_":{}},"msg":"watch stream errored","v":1}
{"level":50,"time":1589484427525,"pid":3746,"hostname":"mx","name":"hyperdrive","component":"drive-manager","id":4,"path":"/","err":{"wrappers_":null,"arrayIndexOffset_":-1,"array":[],"pivot_":1.7976931348623157e+308,"convertedPrimitiveFields_":{}},"msg":"watch stream errored","v":1}
{"level":50,"time":1589484427525,"pid":3746,"hostname":"mx","name":"hyperdrive","component":"drive-manager","id":4,"path":"/","err":{"wrappers_":null,"arrayIndexOffset_":-1,"array":[],"pivot_":1.7976931348623157e+308,"convertedPrimitiveFields_":{}},"msg":"watch stream errored","v":1}
{"level":50,"time":1589484467020,"pid":3746,"hostname":"mx","name":"hyperdrive","component":"drive-manager","id":4,"path":"/","err":{"wrappers_":null,"arrayIndexOffset_":-1,"array":[],"pivot_":1.7976931348623157e+308,"convertedPrimitiveFields_":{}},"msg":"watch stream errored","v":1}
{"level":30,"time":1589485105397,"pid":3746,"hostname":"mx","name":"hyperdrive","component":"drive-manager","key":"2993fc00bf5632a53c85423cc8eea4396791a5a0d0a702e8ef9c79aa8cd63b94","msg":"registering mountpoint in drive index","v":1}
```

**Comment by danrobi11** [comment]
@pfrazee 'Does the editor work?'
Yes works here with new project

**Comment by pfrazee** [comment]
So I think the error is actually not "File Not Found" but beaker is mis-translating it (known issue). That's making it harder for us to diagnose. Is there anything interesting in `beaker://settings/?view=fs-audit-log`?

**Comment by danrobi11** [comment]
First thing im noticing is that the scroll-bar not working as intented in  beaker://settings/?view=fs-audit-log. Wont let me scroll the entire page only a small protion of it.  make it really hard to find anything cuz I need to use the scroll-wheel and that lod does not end.
I can open separate issue for that if u want.

**Comment by pfrazee** [comment]
Nah I put a todo for myself. Try clicking in the log-table.

**Comment by rakoo** [comment]
Another linux user here, using the provided AppImage on Archlinux and I have the same issue when trying to edit my profile: the edit window is blank, all I see is the "Select a file" link.

Now after clicking around a few times it seems something has broken, it didn't want to kill and after a forced shutdown it doesn't want to start anymore (blocked on the "Initializing Beaker" message).

I know I don't provide much to help you with, but I hope you'll have enough information to understand what's happening

Thanks for the project btw !

**Comment by theorangepotato** [comment]
For me, the File Explorer only shows `Error: Uncaught NotFoundError: File not found`, and while the editor will show the list of files, it will not show me their plaintext content so that I can edit them. The log includes another error like this:
```
error:    Uncaught exception: TypeError: cb is not a function {"category":"browser"}
2020-05-14T14:34:29: PM2 error: Uncaught exception: TypeError: cb is not a function
    at /tmp/.mount_Beaker0JxJD1/resources/app/node_modules/call-me-maybe/index.js:11:28
    at processTicksAndRejections (internal/process/task_queues.js:76:11)
```
When I click on the Terminal, the web terminal is blank, and it spams the above error in the bash terminal I launched Beaker from until I close the tab.

My `.hyperdrive/log.json` has 18 messages of
```json
{"level":50,"time":1589492061881,"pid":25038,"hostname":"host","name":"hyperdrive","component":"drive-manager","id":2,"path":"/","err":{"wrappers_":null,"arrayIndexOffset_":-1,"array":[],"pivot_":1.7976931348623157e+308,"convertedPrimitiveFields_":{}},"msg":"watch stream errored","v":1}
```
with different time stamps, but these errors do not seem to be affected by the above actions.

**Comment by danrobi11** [comment]
@rakoo run this in terminal and try to launch the appimage after, might work.
- sudo sysctl kernel.unprivileged_userns_clone=1

**Comment by pfrazee** [comment]
Okay this is helping, thank you everybody. I think the common thread here is AppImage on Linux, though distro doesn’t seem to matter. That “cb is not a function” is probably also a good clue.

Please feel free to add more information. We’re prioritizing the bug (though we’re pretty wiped so it’ll be tomorrow).

**Comment by pfrazee** [comment]
This is a weak solution but if the problem is actually AppImage, and you want to get past this issue, it's not _terribly_ hard to build from source until we solve this. Instructions here: https://github.com/beakerbrowser/beaker#building-from-source

**Comment by rakoo** [comment]
Indeed I used the install instruction and it works as expected, I can see the editor :+1:

**Comment by apederson94** [comment]
Just ran across this issue as well. building from source did not initially work. However, I was able to solve my issue with the building by running `sudo apt install python` on Ubuntu 20.04. The error I was getting was tied to sqlite3 during the `npm rebuild` command.

Hopefully this helps someone else!

**Comment by yarko** [comment]
So, I had trouble; I just built from source, and am consistently getting this (Ubuntu 18.04;  if it matters, I build my own python's):

```
yarko@yarko-Swift-SF314-51:~/workspace/beakerbrowser/beaker/scripts$ npm start

> beakerbrowser@ start /home/yarko/workspace/beakerbrowser/beaker/scripts
> gulp start

[17:28:56] Using gulpfile ~/workspace/beakerbrowser/beaker/scripts/gulpfile.js
[17:28:56] Starting 'start'...
[17:28:56] Starting '<anonymous>'...
Spawning electron /home/yarko/workspace/beakerbrowser/beaker/scripts/node_modules/electron/dist/electron
App threw an error during load
Error: Cannot find module 'hyperdrive-daemon'
Require stack:
- /home/yarko/workspace/beakerbrowser/beaker/app/main.build.js
- /home/yarko/workspace/beakerbrowser/beaker/scripts/node_modules/electron/dist/resources/default_app.asar/main.js
- 
    at Module._resolveFilename (internal/modules/cjs/loader.js:797:17)
    at Function../lib/common/reset-search-paths.ts.Module._resolveFilename (electron/js2c/browser_init.js:7735:16)
    at Module._load (internal/modules/cjs/loader.js:690:27)
    at Module._load (electron/js2c/asar.js:738:28)
    at Function.Module._load (electron/js2c/asar.js:738:28)
    at Module.require (internal/modules/cjs/loader.js:852:19)
    at require (internal/modules/cjs/helpers.js:74:18)
    at /home/yarko/workspace/beakerbrowser/beaker/app/main.build.js:49:40
    at Object.<anonymous> (/home/yarko/workspace/beakerbrowser/beaker/app/main.build.js:16488:2)
    at Module._compile (internal/modules/cjs/loader.js:967:30)
Gtk-Message: 17:28:57.319: GtkDialog mapped without a transient parent. This is discouraged.
```

Running `npm run burnthemall` didn't seem to have any effect.

**Comment by yarko** [comment]
Looking at @pfrazee's repos, I followed instructions on `https://github.com/hypercore-protocol/hyperdrive-daemon` and ran `npm i hyperdrive-daemon -g`

If it helps / gives any hint, this came from it:

```
> leveldown@5.6.0 install /usr/lib/node_modules/hyperdrive-daemon/node_modules/leveldown
> node-gyp-build

npm WARN optional SKIPPING OPTIONAL DEPENDENCY: fsevents@~2.1.2 (node_modules/hyperdrive-daemon/node_modules/chokidar/node_modules/fsevents):
npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for fsevents@2.1.3: wanted {"os":"darwin","arch":"any"} (current: {"os":"linux","arch":"x64"})
```

**Comment by okdistribute** [comment]
I was able to fix this today by removing the `~/.config/Beaker Browser` directory.

**Comment by aadilayub** [comment]
@okdistribute's solution worked for me as well (although I'm not using the AppImage; i installed from source)

**Comment by johanricher** [comment]
If I delete the  `~/.config/Beaker Browser` directory, it's recreated when I run Beaker again, and the "Error: Uncaught NotFoundError: File not found" bug is still there.
(Firefox, Ubuntu 20.04)

**Comment by pfrazee** [comment]
So I _think_ this is being caused by an issue with the built AppImage that was caused by a problem in our build scripts. (At least, I'm very sure that caused the "cb is not a function" error.)

I've uploaded a "prerelease.1-and-a-half" AppImage here: https://github.com/beakerbrowser/beaker/releases/download/1.0.0-prerelease.1/Beaker.Browser-1.0.0-prerelease.1-and-a-half.AppImage

Give that a try and let me know if that solves it.

**Comment by theorangepotato** [comment]
> prerelease.1-and-a-half

After some basic testing around, it does seem to have fixed the issue for me!

**Comment by pfrazee** [comment]
Okay great, pretty sure this is solved. LMK if the fixed AppImage doesn't work.

**Comment by johanricher** [comment]
Works for me too. :+1: Thank you @pfrazee!

**Comment by thecherrybomb** [comment]
Well, looks like it works.
Wouldn't let me save my old profile after editing, but after I deleted the ~/.config/Beaker Browser folder, it did.

**Comment by dym-sh** [comment]
tried clean install of AppImage: 
```
Failed to connect to daemon, retrying [info]
{ "category": "hyper", "subcategory": "daemon", "message": "Failed to connect to daemon, retrying", "level": "info", "timestamp": "2020-05-20T15:32:59.985Z", "type": "data", "target": {} }
```

updated files from git (several days of commits) – 
_now it has the same problem_ :face_with_head_bandage: 
(yes, cleaned things from `~/` and also clean git-clone)

**Comment by pfrazee** [comment]
@dym-sh That error isn't actually a failure, the daemon will sometimes require multiple connection attempts because it takes some time to start

**Comment by dym-sh** [comment]
![PrtScr--2020-05-20--180543](https://user-images.githubusercontent.com/59403389/82469681-a7573800-9ac4-11ea-9ab3-4fb3320d42f2.png)

**Comment by pfrazee** [comment]
Ah well that is a failure. Anything interesting in `~/.hyperdrive/log.json`? Any zombie processes named 'beaker' or 'hyperdrive' ?

**Comment by dym-sh** [comment]
ehm, log size is zero

**Comment by yarko** [comment]
I can now get past "creating a user profile" with 1-and-a-half appimage;
- Next issue:
beaker.user.directory:
   "sign in via twitter" is in a seemingly endless loop (I have 2FA on twitter).
  back up to initial screen;
  return to user directory screen; auth beaker for twitter works 2nd time;
  "add your listing" - fails 
![beaker from 2020-05-20 14-43-12](https://user-images.githubusercontent.com/36553/82490076-520d2d80-9aa8-11ea-8739-f30e94ea506f.png)

Returning (back button) from user listing:
`Failed to find request token in session`

Kill tab (back to "Welcome to Beaker" page / tab):
and "add yourself to directory" (again) shows me logged in via twitter, but cannot add (as above, still);

**Comment by yarko** [comment]
Update:   last "invocation" was from shell command line;
Killed beaker, restarted from GTK GUI, and:
- asked to re-create identity;
- progress to "add yourself to directory":
  - twitter auth already active;
  - this time, my profile appears in popup for "add yourself to the directory",  but:
    - profile name is not same as twitter name;
    - directory listing appears to ignore profile name;

**Comment by pfrazee** [comment]
@dym-sh so we ran into some issues with a GRPC update (which only showed itself to us in Windows) so we had to downgrade it. Try the prerelease.2 appimage we're about to release.

One other thing I've seen on occasion is a situation where the daemon will, at random, fail to write to a profile drive as if the private key is missing. We've got a TODO to figure that out.

@yarko that sounds like a whole different can of worms, I'd suggest opening a separate issue.

**Comment by yarko** [comment]
@pfrazee  - I'll wait on the prerelease.2 appimage, and start from there...

**Comment by dym-sh** [comment]
@pfrazee ok, there was a hyperdrive process leftover i missed that;
current master build works, i'll follow-back if new appimage gonna have issues

**Comment by dym-sh** [comment]
had to
``` sh
chmod +x Beaker.Browser-1.0.0-prerelease.2.AppImage
sudo sysctl kernel.unprivileged_userns_clone=1
```
to make appimage work on linux, but it seems stable

**Comment by coorey** [comment]
I was experiencing this issue. Using prerelease.2 appimage seems to fix it.
